[
  {
    "cmd": [],
    "name": "helper",
    "~followup_annotations": [
      "@@@STEP_FAILURE@@@"
    ]
  },
  {
    "cmd": [
      "python3",
      "RECIPE[recipe_engine::futures:examples/background_helper].resources/helper.py",
      "[CLEANUP]/pid_file"
    ],
    "cost": {
      "cpu": 0,
      "disk": 0,
      "memory": 0,
      "net": 0
    },
    "name": "helper.helper loop",
    "~followup_annotations": [
      "@@@STEP_NEST_LEVEL@1@@@"
    ]
  },
  {
    "cmd": [
      "python3",
      "RECIPE[recipe_engine::futures:examples/background_helper].resources/wait_for_helper.py",
      "[CLEANUP]/pid_file",
      "/path/to/tmp/json"
    ],
    "cost": {
      "cpu": 0,
      "disk": 0,
      "memory": 0,
      "net": 0
    },
    "name": "helper.wait for it",
    "timeout": 30,
    "~followup_annotations": [
      "@@@STEP_NEST_LEVEL@1@@@",
      "@@@STEP_LOG_END@json.output (invalid)@@@",
      "@@@STEP_LOG_LINE@json.output (exception)@Expecting value: line 1 column 1 (char 0)@@@",
      "@@@STEP_LOG_END@json.output (exception)@@@",
      "@@@STEP_FAILURE@@@"
    ]
  },
  {
    "failure": {
      "failure": {},
      "humanReason": "timed out while waiting for helper"
    },
    "name": "$result"
  }
]